core/src/execution/datafusion/planner.rs

```diff
@@ -978,7 +979,15 @@ impl PhysicalPlanner {
                     // `EqualNullSafe`, Spark will rewrite it during planning.
                     false,
                 )?);
-                Ok((scans, join))
+
+                // If the hash join is build right, we need to swap the left and right
+                let hash_join = if join.build_side == 0 {
```

Comment:
nit: we can use the generated enum from the proto here to reference `BuildLeft` rather than `0`.